core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala

```diff
@@ -0,0 +1,214 @@
+            error("ZooKeeper session expired while changing ACLs")
+            exception = throw ZkException.create(KeeperException.create(Code.get(rc)))
+          case _ =>
+            error("Unexpected return code: %d".format(rc))
+            exception = throw ZkException.create(KeeperException.create(Code.get(rc)))   
+      }
+    }
+  }
+  
+  def outputUsage(parser: OptionParser) {
+    CommandLineUtils.printUsageAndDie(parser, "ZooKeeper Migration Tool Help. This tool updates the ACLs of "
+                                                + "znodes as part of the process of setting up ZooKeeper "
+                                                + "authentication.")
+  }
+  
+  def run(args: Array[String]) {
+    var jaasFile = System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM)
+    val parser = new OptionParser()
+
+    val goOpt = parser.accepts("kafka.go", "Indicates whether to make this cluster secure or unsecure. The "
+        + " options are 'secure' and 'unsecure'").withRequiredArg().ofType(classOf[String])
+    val jaasFileOpt = parser.accepts("jaas.file", "JAAS Config file.").withOptionalArg().ofType(classOf[String])
+    val zkUrlOpt = parser.accepts("zookeeper.connect", "Sets the ZooKeeper connect string (ensemble). This parameter " +
+      "takes a comma-separated list of host:port pairs.").withRequiredArg().defaultsTo("localhost:2181").
+      ofType(classOf[String])
+    val zkSessionTimeoutOpt = parser.accepts("zookeeper.session.timeout", "Sets the ZooKeeper session timeout.").
+      withRequiredArg().defaultsTo("30000").ofType(classOf[java.lang.Integer])
+    val zkConnectionTimeoutOpt = parser.accepts("zookeeper.connection.timeout", "Sets the ZooKeeper connection timeout.").
+      withRequiredArg().defaultsTo("30000").ofType(classOf[java.lang.Integer])
+    val helpOpt = parser.accepts("help", "Print usage information.")
+
+    val options = parser.parse(args : _*)
+    if(options.has(helpOpt))
+      outputUsage(parser)
+
+    if ((jaasFile == null) && !options.has(jaasFileOpt)) {
+      error("No JAAS configuration file has been specified. Please make sure that you have set either "
+            + "the system property %s or the option %s".format(JaasUtils.JAVA_LOGIN_CONFIG_PARAM, "--jaas.file"))
+      throw new IllegalArgumentException("Incorrect configuration")
+    }
+    
+    if (jaasFile == null) {
+      jaasFile = options.valueOf(jaasFileOpt)
+      System.setProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM, jaasFile)
+    }
+    
+    if (!JaasUtils.isZkSecurityEnabled(jaasFile)) {
+      error("Security isn't enabled, most likely the file isn't set properly: %s".format(jaasFile))
+      throw new IllegalArgumentException("Incorrect configuration") 
+    }
+
+    var goSecure: Boolean = false
+    val go = options.valueOf(goOpt) match {
+      case "secure" =>
+        goSecure = true
+      case "unsecure" =>
+        goSecure = false
```

Comment:
Scala is expression-based instead of statement-based, you can just do something like:

```scala
val goSecure = foo match {
  case "secure" => true
  case "unsecure" => false
  case _ => outputUsage(parser)
````

But you also need to change `outputParser` as I explained in that line.